### PR TITLE
`sage.numerical`: Update `# needs`

### PR DESCRIPTION
**Pull request by mkoeppe**
Also removing a function deprecated in #32226

- In part cherry picked from #35095 
- Part of #29705 

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/numerical/backends/generic_backend.pyx [hunk 1]
```diff
@@ -123,28 +124,30 @@ cdef class GenericBackend:
...
             5
-            sage: p.add_variables(2, lower_bound=-2.0, integer=True, names=['a','b']) # optional - Nonexistent_LP_solver
+            sage: p.add_variables(2, lower_bound=-2.0, integer=True, names=['a','b'])
             6
 
         TESTS:
 
         Check that arguments are used::
 
-            sage: p.col_bounds(5) # tol 1e-8, optional - Nonexistent_LP_solver
```
Comment (dcoudert) [inline src/sage/numerical/backends/generic_backend.pyx:141]: you also removed the `# too 1e-8` tag. Is it safe ?

File: src/sage/numerical/backends/generic_backend.pyx [hunk 2]
```diff
@@ -123,28 +124,30 @@ cdef class GenericBackend:
...
+            sage: p.col_bounds(5)
             (-2.0, None)
-            sage: p.is_variable_integer(5)   # optional - Nonexistent_LP_solver
+            sage: p.is_variable_integer(5)
             True
-            sage: p.col_name(5)              # optional - Nonexistent_LP_solver
+            sage: p.col_name(5)
             'a'
-            sage: p.objective_coefficient(5) # tol 1e-8, optional - Nonexistent_LP_solver
+            sage: p.objective_coefficient(5)
```
Comment (dcoudert) [inline src/sage/numerical/backends/generic_backend.pyx]: same here

**Review by dcoudert — COMMENTED** [review]
otherwise LGTM. Passes all tests without warnings on Fedora 35.

File: src/sage/numerical/backends/generic_backend.pyx (hunk 1 above)
Comment (mkoeppe) [reply src/sage/numerical/backends/generic_backend.pyx:141]: Thanks for spotting this unintended change. Fixed in 314d772

**Review by dcoudert — APPROVED** [review]
So then, LGTM.

**Comment by mkoeppe** [comment]
Thank you!

**Comment by mkoeppe** [comment]
rebased

**Pull request merged**